pkg/snapstore/snapstore_test.go

```diff
@@ -649,13 +650,41 @@ var _ = Describe("Server Side Encryption Customer Managed Key for S3", func() {
 	})
 })
 
-var _ = Describe("Get Immutability time for S3 bucket", func() {
+var _ = Describe("Get Bucket versioning status for S3 buckets", func() {
 	awsS3Client := &mockS3Client{
 		objects:          objectMap,
 		prefix:           prefixV2,
 		multiPartUploads: map[string]*[][]byte{},
 	}
 	Context("S3 bucket with object lock enabled", func() {
+		It("Should return enabled versioning status", func() {
+			versioningStatus, err := awsS3Client.GetBucketVersioning(&s3.GetBucketVersioningInput{
+				Bucket: &s3ObjectLockedBucket,
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(*versioningStatus.Status).Should(Equal(s3.BucketVersioningStatusEnabled))
+		})
+	})
+	Context("S3 bucket with object lock not enabled", func() {
+		It("Should return versioning status as nil", func() {
+			versioningStatus, err := awsS3Client.GetBucketVersioning(&s3.GetBucketVersioningInput{
+				Bucket: &s3NonObjectLockedBucket,
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(versioningStatus.Status).Should(BeNil())
+		})
+	})
+})
+
+var _ = Describe("Get Immutability time for S3 bucket", func() {
+	awsS3Client := &mockS3Client{
+		objects:          objectMap,
+		prefix:           prefixV2,
+		multiPartUploads: map[string]*[][]byte{},
+	}
+	var s3ObjectLockBucketBUTRulesNotDefined = "mock-s3ObjectLockBucketBUTRulesNotDefined"
```

Comment:
Can you please add a comment about what `s3ObjectLockBucketBUTRulesNotDefined` exactly means here?

Reply:
Ah then can you please rename to `s3ObjectLockBucketButRulesNotDefined`. The `BUT` was capitalized, so that threw me off, since I thought it was an abbreviation for something 😅

Reply:
ok, no worries, I will rename it.